crates/pop-common/src/manifest.rs

```diff
@@ -91,6 +92,89 @@ pub fn add_crate_to_workspace(workspace_toml: &Path, crate_path: &Path) -> anyho
 	Ok(())
 }
 
+/// Collects the dependencies of the given `Cargo.toml` manifests in a HashMap.
+///
+/// # Arguments
+/// * `manifests`: Paths of the manifests to collect dependencies from.
+pub fn collect_manifest_dependencies(manifests: Vec<&Path>) -> anyhow::Result<HashSet<String>> {
+	let mut dependencies = HashSet::new();
+	for m in manifests {
+		let cargo = &std::fs::read_to_string(m)?.parse::<DocumentMut>()?;
+		for d in cargo["dependencies"].as_table().unwrap().into_iter() {
+			dependencies.insert(d.0.into());
+		}
+		for d in cargo["build-dependencies"].as_table().unwrap().into_iter() {
+			dependencies.insert(d.0.into());
+		}
+	}
+	Ok(dependencies)
+}
+
+/// Extends `target` cargo manifest with `dependencies` using versions from `source`.
+///
+/// It is useful when the list of dependencies is compiled from some external crates,
+/// but there is interest on using a separate manifest as the source of truth for their versions.
+///
+/// # Arguments
+/// * `target`: Manifest to be modified.
+/// * `source`: Manifest used as reference for the dependency versions.
+/// * `template`: The name of the template for the target manifest.
+/// * `tag`: Version to use when transforming local dependencies from `source`.
+/// * `dependencies`: List to extend `target` with.
+pub fn extend_dependencies_with_version_source<I>(
+	target: &mut Manifest,
+	source: Manifest,
+	template: String,
+	tag: Option<String>,
+	dependencies: I,
+) where
+	I: IntoIterator<Item = String>,
+{
+	let mut target_manifest_workspace = target.workspace.clone().unwrap();
+	let source_manifest_workspace = source.workspace.clone().unwrap();
+
+	let updated_dependencies: Vec<_> = dependencies
+		.into_iter()
+		.filter_map(|k| {
+			if let Some(dependency) = source_manifest_workspace.dependencies.get(&k) {
+				if let Some(d) = dependency.detail() {
+					let mut detail = d.clone();
+					if d.path.is_some() {
+						detail.path = None;
+						detail.git = Some("https://github.com/moondance-labs/tanssi".to_string());
```

Comment:
Since this is a public library on crates.io intended for anyone to integrate, it shouldn't contain hardcoded values like this. Instead, the value should be passed as a parameter, and have any hardcoded values only in the `pop-cli` crate.

Reply:
That makes complete sense.

Reply:
This has now been addressed in my last commit.